src/Components/getList.js

```diff
@@ -27,16 +27,67 @@ function EmptyList() {
   );
 }
 
-function FullList(props) {
+
+const FullList = (props)=> {
   const [isChecked, setCheck] = useState(false);
+  const [searchTerm, setSearchTerm] = useState("");
+
+  const handleChange = e => {
+    setSearchTerm(e.target.value);
+  };
+
+  const clearSearch = () => {
+    setSearchTerm("");
+  };
+
+  const searchList = props.data.filter(item => {
+    return item.itemName.toLowerCase().includes(searchTerm.toLowerCase());
+  });
+
 
 
   return (
     <div className="grocery-list">
       <h1>Groceries</h1>
-      <ul className="tl f2">
-        {props.data.map(item => (
+      <ul className="tl f2 check-list">
+      <li className="bg-white pa1 shadow">
+      <div className="filter-input">
+      <input 
+         type="text"
+         placeholder="Filter List"
+         value={searchTerm}
+         onChange={handleChange}
+       />
+       <button className="filter-button" onClick={clearSearch}> X </button>
+       </div>
+       </li>
+
+    
+        {searchTerm
+          ? searchList.map(item => (
+            <li className="bg-white pa1 shadow" key={item.id}>
```

Comment:
For ease of maintainability of this file, you could create a new React component to express this `<li>`, and reuse it for both cases:

```js
searchTerm
  ? searchList.map(item => <GroceryItem item={item} />)
  : props.data.map(item => <GroceryItem item={item} />)

function GroceryItem(props) {
  const { item } = props;
  return (
    <li className="bg-white pa1 shadow" key={item.id}>
      <div className="checkbox-input">
        <label>
          <input
            type="checkbox"
            value={isChecked}
            onChange={() => setCheck(checked => !checked)}
            defaultChecked={
              item.lastPurchasedDate
                ? Date.now() / 1000 - item.lastPurchasedDate.seconds < 86400
                : false
            }
          />
          {item.itemName}
        </label>
      </div>
    </li>
  );
}
```

Reply:
👏 Agree 100%

Reply:
This was one of our goals but we were getting a little lost in how the props could be accessed from function to function.  This is super helpful!